test_bot/test_bot.py

```diff
@@ -297,12 +297,8 @@ def engine_path(CONFIG: CONFIG_DICT_TYPE) -> str:
         name: str = CONFIG["engine"]["name"]
         return os.path.join(dir, name)
 
-    if platform == "win32":
-        CONFIG["engine"]["name"] = "buggy_engine.bat"
-    else:
-        CONFIG["engine"]["name"] = "buggy_engine"
-        st = os.stat(engine_path(CONFIG))
-        os.chmod(engine_path(CONFIG), st.st_mode | stat.S_IEXEC)
+    CONFIG["engine"]["name"] = "buggy_engine.py"
```

Comment:
The `buggy_engine` is used for both sides of the game. The later call to `engine_path(CONFIG)` on line 305 fails to get the right name for the executable due to the change in this line. Within `engine_path()`, the `name` variable needs to be either `buggy_engine.bat` or `buggy_engine` depending on the `platform` variable. Right now, it returns `buggy_engine.py`, which can't be run by directly calling it.

Reply:
Here's what I came up with. There's probably better ways.
```python
    def engine_path(CONFIG: CONFIG_DICT_TYPE) -> str:
        dir: str = CONFIG["engine"]["dir"]
        name: str = CONFIG["engine"]["name"].removesuffix(".py")
        if platform == "win32":
            name = name + ".bat"
        return os.path.join(dir, name)
```